src/leveleditor/leveleditor.cpp

```diff
@@ -2327,14 +2327,26 @@ int editor_platforms()
 						}
                 } else if (event.key.keysym.sym == SDLK_t) {
                     if (PLATFORM_EDIT_STATE_EDIT == iPlatformEditState || PLATFORM_EDIT_STATE_ANIMATED == iPlatformEditState || PLATFORM_EDIT_STATE_TILETYPE == iPlatformEditState) {
-							editor_tiles();
+							FPSLimiter::instance().frameStart();
+							while (editor_tiles() == EDITOR_TILES) {
+								FPSLimiter::instance().beforeFlip();
+								gfx_flipscreen();
+								FPSLimiter::instance().afterFlip();
+								FPSLimiter::instance().frameStart();
+							}
```

Comment:
Is there any chance `editor_tiles()` will not return `EDITOR_TILES` the first time it's called? That case would cause the rest of the fps limiter methods to be not called either.

Reply:
Yes, if you close the tileset menu with frame perfect timing, but that doesn't matter because you don't really need to draw the menu if you close it.

Reply:
:+1: